test/bigquery.rb

```diff
@@ -120,6 +120,19 @@ def test_for_query
     assert_equal result['jobComplete'], true
   end
 
+  def test_for_query_useQueryCache
+    result = @bq.query("SELECT * FROM [#{config['dataset']}.test] LIMIT 1", useQueryCache: true)
+    result = @bq.query("SELECT * FROM [#{config['dataset']}.test] LIMIT 1", useQueryCache: true)
+
+    assert_equal result['cacheHit'], true
+  end
+
+  def test_for_query_dryRun
+    result = @bq.query("SELECT * FROM [#{config['dataset']}.test] LIMIT 1", dryRun: true)
+
+    assert_equal result['jobReference']['jobId'], nil
+  end
+
```

Comment:
I could not write a test for `maxResults` because my test dataset had had only 1 row data...